cmd/zed/zed.d/statechange-led.sh

```diff
@@ -122,54 +139,66 @@ process_pool()
 
 	# The output will be the vdevs only (from "grep '/dev/'"):
 	#
-	#    U45     ONLINE       0     0     0   /dev/sdk          0
-	#    U46     ONLINE       0     0     0   /dev/sdm          0
-	#    U47     ONLINE       0     0     0   /dev/sdn          0
-	#    U50     ONLINE       0     0     0  /dev/sdbn          0
+	#    U45     ONLINE       0     0     0   /dev/sdk          0       0
+	#    U46     ONLINE       0     0     0   /dev/sdm          0       0
+	#    U47     ONLINE       0     0     0   /dev/sdn          0       0
+	#    U50     ONLINE       0     0     0   Potential info message /dev/sdbn          0       0
 	#
-	ZPOOL_SCRIPTS_AS_ROOT=1 $ZPOOL status -c upath,fault_led "$pool" | grep '/dev/' | (
+	ZPOOL_SCRIPTS_AS_ROOT=1 $ZPOOL status -c upath,fault_led,locate_led "$pool" | grep '/dev/' | (
 	rc=0
 	while read -r vdev state _ _ _ therest; do
-		# Read out current LED value and path
+		# Read out current LED values and path
 		# Get dev name (like 'sda')
-		dev=$(basename "$(echo "$therest" | awk '{print $(NF-1)}')")
+		dev=$(basename "$(echo "$therest" | awk '{print $(NF-2)}')")
 		vdev_enc_sysfs_path=$(realpath "/sys/class/block/$dev/device/enclosure_device"*)
-		current_val=$(echo "$therest" | awk '{print $NF}')
-
-		if [ "$current_val" != "0" ] ; then
-			current_val=1
-		fi
-
 		if [ -z "$vdev_enc_sysfs_path" ] ; then
 			# Skip anything with no sysfs LED entries
 			continue
 		fi
 
-		if [ ! -e "$vdev_enc_sysfs_path/fault" ] ; then
-			rc=3
-			zed_log_msg "vdev $vdev '$file/fault' doesn't exist"
-			continue
-		fi
+		fault_val=$(echo "$therest" | awk '{print $(NF-1)}')
+		locate_val=$(echo "$therest" | awk '{print $NF}')
 
-		val=$(state_to_val "$state")
-
-		if [ "$current_val" = "$val" ] ; then
-			# LED is already set correctly
-			continue
+		# Treat all non-zero values as 1
```

Comment:
The entire hunk up to here could be better served as 
```sh
	# The output will be the vdevs only (by matching "/dev/"):
	#
	#    U45     ONLINE       0     0     0   /dev/sdk          0       0
	#    U46     ONLINE       0     0     0   /dev/sdm          0       0
	#    U47     ONLINE       0     0     0   /dev/sdn          0       0
	#    U50     ONLINE       0     0     0   Potential info message /dev/sdbn          0       0
	#
	ZPOOL_SCRIPTS_AS_ROOT=1 $ZPOOL status -c upath,fault_led,locate_led "$pool" | awk '/\/dev\// {print $1 " " $2 " " $(NF-2) " " $(NF-1) " " $NF}' | (
	rc=0
	while read -r vdev state dev fault_val locate_val; do
		# Read out current LED values and path
		# Get dev name (like 'sda')
		dev=$(basename "$dev")
		vdev_enc_sysfs_path=$(realpath "/sys/class/block/$dev/device/enclosure_device"*)
		if [ -z "$vdev_enc_sysfs_path" ] ; then
			# Skip anything with no sysfs LED entries
			continue
		fi

		# Treat all non-zero values as 1
		...
```

This avoids an AWK per variable

Reply:
In a271946d724ae1d399564ace759fce8e00686812 but superseeded by e561d8d0ec365fc02b78b22388041ead081a5957